booking-service/src/main/java/fr/xebia/xke/micronaut/booking/domain/Quantity.java

```diff
@@ -18,8 +18,13 @@ public Quantity(@NonNull final Long value) {
         this.value = value;
     }
 
-    public Quantity add(@NonNull final Quantity quantity) {
-        return new Quantity(this.value + quantity.value);
+    public Quantity subtract(@NonNull final Quantity quantity) {
+        checkArgument(this.isGreaterThan(quantity),
+                "Cannot subtract more than available quantity (available quantity: %s, requested quantity: %s)", this, quantity);
+        return new Quantity(this.value - quantity.value);
     }
 
+    public boolean isGreaterThan(final Quantity quantity) {
+        return quantity.value > this.value;
```

Comment:
i am pretty sure something is wrong here!
`this.isGreaterThan(quantity)` should return `this.value > quantity.value`
make sure to fix the tests before you fix the implementation, the fact that they did not expose this issue is an issue in itself.